### PR TITLE
FIX: Add Retry-Header to rate limited responses

### PR DESCRIPTION
**Pull request by udan11**
It returned a 429 error code with a 'Retry-After' header if a
RateLimiter::LimitExceeded was raised and unhandled, but the header was
missing if the request was limited in the 'RequestTracker' middleware.

### CODE REVIEW

File: lib/middleware/request_tracker.rb [hunk 1]
```diff
@@ -166,8 +166,8 @@ def call(env)
 
     request = Rack::Request.new(env)
 
-    if rate_limit(request)
-      result = [429, {}, ["Slow down, too many requests from this IP address"]]
+    if available_in = rate_limit(request)
+      result = [429, { "Retry-After" => available_in }, ["Slow down, too many requests from this IP address"]]
       return result
```
Comment (ZogStriP) [inline lib/middleware/request_tracker.rb:171]: Why not combine these two lines?

```suggestion
      return [429, { "Retry-After" => available_in }, ["Slow down, too many requests from this IP address"]]
```

File: lib/middleware/request_tracker.rb [hunk 2]
```diff
@@ -273,7 +273,7 @@ def rate_limit(request)
           Discourse.warn("Global asset IP rate limit exceeded for #{ip}: 10 second rate limit", uri: request.env["REQUEST_URI"])
         end
 
-        return !(GlobalSetting.max_reqs_per_ip_mode == "warn")
+        return GlobalSetting.max_reqs_per_ip_mode != "warn" && limiter_assets10.seconds_to_wait(Time.now.to_i)
```
Comment (ZogStriP) [inline lib/middleware/request_tracker.rb:276]: That looks a bit weird. Hard to tell what `limiter_assets10.seconds_to_wait` returns and when is it `true` or `false`

File: lib/middleware/request_tracker.rb [hunk 3]
```diff
@@ -282,12 +282,12 @@ def rate_limit(request)
         type = 60
         limiter60.performed!
         false
-      rescue RateLimiter::LimitExceeded
+      rescue RateLimiter::LimitExceeded => e
         if warn
           Discourse.warn("Global IP rate limit exceeded for #{ip}: #{type} second rate limit", uri: request.env["REQUEST_URI"])
-          !(GlobalSetting.max_reqs_per_ip_mode == "warn")
+          GlobalSetting.max_reqs_per_ip_mode != "warn" && e.available_in
```
Comment (ZogStriP) [inline lib/middleware/request_tracker.rb:288]: Ditto ^^. What does `available_in` mean ?

File: lib/middleware/request_tracker.rb (hunk 2 above)
Comment (udan11) [reply lib/middleware/request_tracker.rb:276]: It should never be `true`. It's either `false` or a number representing the seconds until the request would no longer be blocked.

File: lib/middleware/request_tracker.rb (hunk 3 above)
Comment (udan11) [reply lib/middleware/request_tracker.rb:288]: Applied the same change as before. `available_in` is the name of the field from the `RateLimiter::LimitExceeded` exception which holds the number of seconds until the request is no longer blocked.

**Review by ZogStriP — APPROVED** [review]

**Pull request merged**